### DISCUSSION TITLE
Add agentos.Component.run(entry_point: str, params: dict) -> agentos.Run

### DISCUSSION THREAD

**Discussion by andyk** [agentos-project/agentos#191] [Ideas]
The idea here is to add a `run()` function to `agentos.Component`.

Signature would be something like: `agentos.Component.run(entry_point: str, params: dict) -> agentos.Run`

Under the hood, this would do what the CLI `agentos run` command currently does, but via the python API. Then we can make the CLI version just call this function.

Specifically, the function does something like:
```
i = my_component.get_instance(params)
entry_point_fn = getattr(i, entry_point)
entry_point_fn(params[entry_point])
```

**Comment by andyk** [comment]
Actually, this function already exists in the codebase with the name `agentos.Component.call`! 

https://github.com/agentos-project/agentos/blob/262fa003fc8dbacdcca7555050179c578334384b/agentos/component.py#L181-L187

**Comment by andyk** [comment]
> Actually, this function already exists in the codebase with the name agentos.Component.call!

I haven't double checked to see if the CLI `agentos run` command now calls that under the hood or not.

**Reply by nickjalbert** [reply]
Yep, `agentos run` uses this under the hood (at least in my local branch, the change is relatively recent).  The `call` function could also be extended to accept parameters to override any previously-specified ones.

**Reply by andyk** [reply]
I was just writing up a proposal in my in-progress code review of #188 that: 

1. we harmonize the naming of `Component.call` and the CLI `agentos run` command (I'm leaning slightly towards `run` rather than `call`)
2. `Component.call` (or `run` if we rename) take a `params` args


To discuss (after you read my longer thoughts in my code review): should we consider create a core abstraction `agentos.Params` which would be thin wrapper around python dict where the param parsing logic would live (i.e. factor the concept of Params out of the Component concept).
